cx-sig-charter-draft.md

```diff
@@ -0,0 +1,81 @@
+- Establish and share metrics to measure project health, community health, and general trends, including:
+  - [DevStats](https://spinnaker.devstats.cd.foundation/)
+  - the CDF contribution gamification app [[WIP](https://github.com/ExitoLab/spinnaker_gamification_app)]
+  - contributor experience survey(s)
+  - engagement on project platforms
+  - pull request and issue engagement and closure times
+- Set community health goals such as improved PR and issue engagement and closure times, and track progress towards their achievement using established metrics.
+- Research other OSS projects and consult with their leaders about contributor experience topics to make sure we are always delivering value to our contributors
+- Create and maintain subprojects that incentivize community outreach, individual content creation, and welcoming/onboarding of new contributors
+
+#### Cross-project and Externally Facing Processes
+
+We cross-cut all SIGs and working groups to deliver the following processes:
+
+- Deploying Changes:
+ This SIG will create change-deployment policies that balance quick response to the needs of the community and a limit on disruptions for project contributors. We will create policies to socialize and elicit feedback on changes via community channels such as Slack, spinnaker.io, Twitter, mailing lists, SIG meetings, delegating to SIG leads, etc. The amount of notice we provide and the amount of consensus we seek is driven by our estimation of risk. We can estimate risk based on these parameters:
+
+- Low-risk changes impact a small number (<3) of SIGs, working groups, or repositories, do not break existing contributor workflows, and are easy to roll back.
+- High-risk changes impact a large number (>3) of SIGs, working groups, or repositories, break existing contributor workflows and are not easy to roll back.
+- Depending on how wide of an ecosystem change this is, we may also slack, blog, tweet, and use other channels to get the word out.
+- Our standard time box is 72 business hours; however, there may be situations where we need to act quickly but the time period will always be clear and upfront.
+- Encourage automation to improve productivity for contributors where it makes sense, and consult with SIG leads on automation workflows.
+
+If we need funding for any reason, we approach [CDF](https://cd.foundation).
+CDF may contribute funding to our platforms, processes, and/or programs. They do not play a day-to-day operations role and have bestowed that to our community to run as we see fit. Since they do fund some of our initiatives, this means that they hold owner account privileges on certain platforms like Zoom and Slack. The CX SIG will work to ensure that project maintainers are included as admins.
+
+### Out of scope
+
+- Code for the testing and CI infrastructure
```

Comment:
What about code for the in-scope projects and services? I'm not clear on whether or not this SIG will be writing code in support of its initiatives, or looking for other groups to perform the legwork. For example, there's _some_ code around contributorship today, are you planning to take over these scripts?

In my mind, the SIG should be responsible for the scripts and tooling that comes out of this SIG if it applies to the project as a whole, including scripts that may be a part of the governance repo itself (e.g. https://github.com/spinnaker/governance/tree/master/scripts and more).

If this SIG is _not_ responsible for the code and tooling that will support the processes and policies this SIG sets out to create, I have little optimism in the short- or long-term success of this SIG's overall goals.

Reply:
The contribEx SIG will be writing the code for the initiatives specific to contribex. For example, if there is slack automation, various automation around owners for this SIG/repo, that is where this will live and be handled by the contribex team.

Reply:
@robzienert, can you please link to the existing code around contributorship to for the sake of this conversation? I'd love to request that existing artifacts be referenced specifically in this PR.

Reply:
I linked to the code in my original comment.

Reply:
Is that the only such existing asset that you would propose this group should maintain, or are there other assets?

Reply:
I think that's it. All membership management has been a manual affair thus far. Other github actions we are maintaining are covered by other SIGs.